cognite/seismic/protos/v1/seismic_service_datatypes.proto

```diff
@@ -0,0 +1,152 @@
+/**
+Messages that describe data types used by Seismic Datastore in Cognite Data Fusion
+*/
+syntax = "proto3";
+package com.cognite.seismic.v1;
+
+import "google/protobuf/wrappers.proto";
+import "cognite/seismic/protos/types.proto";
+
+message Coordinate {
+    float x = 1;
+    float y = 2;
+}
+
+message LineSelect {
+    oneof direction {
+        int32 iline = 1;
+        int32 xline = 2;
+    }
+}
+
+message LineBasedVolume {
+    LineDescriptor iline = 1;
+    LineDescriptor xline = 2;
+    LineDescriptor z = 3;
+}
+
+message LineRange {
+    google.protobuf.Int32Value from_line = 1;
+    google.protobuf.Int32Value to_line = 2;
+}
+
+/**
+ * A cutout of a seismic store
+ */
+ message Seismic {
+    string id = 1;  // Always present
+    string external_id = 2;
+    string name = 3;
+    string crs = 4;
+    map<string, string> metadata = 5;
+    TextHeader text_header = 6;
+    BinaryHeader binary_header = 7;
+    LineRange line_range = 8;
+    CutoutDef cutout_def = 9;
+    string partition_id = 10;
+    string seismicstore_id = 11;  // Present only if user can view the tracestore
+}
+
+/**
+Represents a seismic store.
+**/
+message SeismicStore {
+    string id = 1;
+    string name = 2;
+    string survey_id = 3;
+    TraceSource location = 4; // Where the traces are stored
+    IngestionSource ingestion_source = 5; // How the file was ingested
+    map<string, string> metadata = 6;
+    File ingested_file = 7; // The info for the original file, if it was ingested as a file
+}
+
+/**
+Represents partitions, used for granular access control
+**/
+message Partition {
+    string id = 1;
+    string external_id = 2;
+    string name = 3;
+    repeated Seismic seismics = 4;
+}
+
+/**
+Specifies the cutout "shape"
+**/
+message CutoutDef {
+    string json = 1;  // Uses the Cutout Definition schema
+}
+
+/**
+Used to identify zero or one objects that may have an external id 
+**/
+message Identifier {
+    oneof handle {
+        string id = 1;
+        string external_id = 2;
+    }
+}
+
+/**
+Used to identify any number of objects, intended for any plural Read* endpoints. Set* endpoints should use Identifier.
+
+Can search by id, name, or prefix.
+**/
+message SearchSpec {
+    oneof findby {
+        string id = 1;
+        // External ids only exist for partitions and seismics.
```

Comment:
Seismic stores are created by the ingestion process, so there will have to be capability to add external id when ingesting, because adding external ids retroactively violates the cogapi. Re: surveys, their "name"s currently are effectively external ids. In order to actually change it, it'd require another migration and I think it'd be better to leave that to a different PR.